### DISCUSSION TITLE
Getting the most out of Breadcrumbs

### DISCUSSION THREAD

**Discussion by SkepticMystic** [SkepticMystic/breadcrumbs#175] [Ideas]
There are a few guides on getting started with Breadcrumbs, plus the wiki, but there aren't many resources on more 'advanced' use cases. Let's use this post to discuss how you get the most out of Breadcrumbs. Thanks to @lkadre for the inspiration :)

**Comment by avtfaciam** [comment]
This is going to be very long but please bear with me. I tried to be as clear as possible for both people starting out and people more familiar with the plugin. **This is just how I personally use Breadcrumbs** and hopefully it can give you come ideas!

Thank you so much to @SkepticMystic for creating the plugin and this topic too! 

#### Index note

In the Trail/Grid settings, you can set a page to act as your Index. In my case, my Index is my Publish Homepage. It's the first page people see when they visit my Publish website and it's where they go back to. I honestly don't have that much use for the Trail except to gain an outsider's perspect: if someone opened my vault for the first time, how would they get to where I am right now? It's also a very nice way of cataloguing your note.

#### Hierarchy note

I started using the plugin when the first Alternative Hierarchy feature launched: the [Hierarchy Note](https://github.com/SkepticMystic/breadcrumbs/wiki/Alternative-Hierarchies#hierarchy-notes). This appears in the settings under Hierarchy Notes. In my opinion it's not sustainable to exclusively use the hierarchy note as the list gets way too long, and you have to manually add things, but I think it's a good starting point, especially if you're still just testing the plugin.

#### Figuring out your hierarchies

Start with what you already have. Even if you subscribe to a flat-structure vault, your notes likely already have some sort of hierarchy. When you notice an existing hierarchy, consider adding it to the list of Hierarchies. If you follow what feels most intuitive to you, you're more likely to stick to it. It shouldn't be a chore and you probably don't need to restructure your entire vault to accommodate Breadcrumbs.

If you use Breadcrumbs, you likely already use the [Dataview plugin](https://github.com/blacksmithgu/obsidian-dataview/). As of v. 1.9.0, you can use [Folder indexes](https://github.com/SkepticMystic/breadcrumbs/wiki/Alternative-Hierarchies#folder-indexes) and, as of v. 1.10.0, you can use [Tag indexes](https://github.com/SkepticMystic/breadcrumbs/wiki/Alternative-Hierarchies#tag-indexes). So you can build Breadcrumbs hierarchies from folders, notes or dataview fields – things you probably already have in some form in your vault.

(A disclaimer: when I discovered Breadcrumbs I did spend an entire afternoon, and maybe the evening and the next day too, playing with the plugin and manually adding hierarchical fields to each of my hundreds of notes. Don't be like me. Please value your time.)

#### Folders > Tags > YAML > Inline

If you're concerned about portability (I know I am), take this nifty and entirely made up hierarchy from most portable way to create breadcrumbs to least portable: Folders > Tags > YAML > Inline. 

Aside from my most top-level notes (the Homepage, the note for >> LAW <<, etc), I almost exclusively use tags and inline links. Tags are intuitive to me and very compact, and inline links are non-obstructive to the general content of the note as well as being explicit, unlike YAML. 

I personally find folders constraining but I think the big pushback has gone a bit too far: they absolutely have their uses, and the "folders are bad" narrative has made some people avoid folders when they would clearly benefit from them. Folders are also the most portable way to see your hierarchies and your files. Just check your system file explorer.

#### Example: Literature and author notes

As part of my template for my ""literature"" notes, there's a dataview inline field for the author, like this: `- [i] Author:: {% if author %}[[{{author}}]]{% endif %}`

(Well, this is in the Readwise export setting, but I believe many people have a similar set up in their literature notes, with `author:: [[name]]`). Breadcrumbs picks up on this inline field and, since I have a hierarchy for "author", it reads that the author is above the literature, and all the other works by the same author are its "siblings". Basically, Breadcrumbs is magic.

I don't create notes for every author or person, but I do create them for people who are important in their field or who I keep coming across from, or who I just have something to say about. When I do, the template includes these:

`- [i] nationality:: tag civ/ `
`- [I] movement:: link to movements`

By creating a hierarchy for movement (an art movement, or a philosophy school of thought), I can check in on all the other people that I've taken notes on that also belong to that movement.

I haven't decided whether it's worth it yet, but something I've been considering is "civilisation" hierarchies to compare people from the same "civilisation" to see how thoughts have changed. It'd be especially interesting as often they critique one another. It's not that it's difficult to set up, only that I already have so many works-in-progress that I should probably hold off on that one.

#### Example: Concept notes (Law)

This is specifically about my notes on Law, which I acknowledge is probably very niche but hopefully it can give you some ideas of how to set up your hierarchies and especially how to work with notes in multiple hierarchies. You don't have to choose one!

The first line in my note for Limited Liability Company (DEL) reads: `#j/US/DEL #t/entity #law/corporate `

This automatically places that note within three different hierarchies: jurisdiction (the "j"), type (the "t"), and topic. These are tags and subtags I already used before the Breadcrumbs plugin and clearly denoted "hierarchies", even if they're not immediately explicit. 

Separating by topic feels the most natural because that's how we're used to categorising things. In school I have subjects like "Corporate Law" and "Constitutional Law", so naturally I want to be able to see all my notes related to this topic easily. Similarly, I have tags and subtags for `#philosophy/scepticism` or `#literature/realism`.

Jurisdiction is also a pretty straightforward hierarchy if you have to deal with multiple jurisdictions like I do. By having the jurisdiction hierarchy, I can easily access all rules that would apply to this note since it's in the same jurisdiction (oversimplifying it a bit, but that's the main purpose). Since it's a different hierarchy, it doesn't care about the topic – whether it's the corporate law of Delaware or the regulatory law of Delaware, it all shows up indiscriminately –, but for me that's *good*, because I want to have an overview of the jurisdiction as a whole to understand how the *system* works. But that's getting into Legal theory.

The hierarchy for types took a bit of figuring out, but for my workflow it is important for the purposes of Comparative Law and more easily navigating different systems. Similarly to before, since it is a different hierarchy altogether, it doesn't care about the topic or the jurisdiction. But the idea is precisely to compare between different types of, in this case, entities in different jurisdictions. How does a LLC in Delaware differ from a LLC in NY? How about from a Sociedade Limitada in Brazil? These are different jurisdictions, but they're the same *type* hierarchy.

Cross-referencing with different topics might sound like it gets messy, but it's eye opening. Another example is look through the `#t/principle` and comparing them. When a judge considers a case they weigh different principles just like this, which is a nice exercise. Or, looking through `#t/authority`, I can consider the American SEC against the Brazilian CVM. 

*And none of these are exclusive.* If I want to focus on one hierarchy, I can fold the others. Breadcrumbs offers me all the potential of hierarchies, and doesn't really take anything away.

**Reply by brimwats1** [reply]
do you have your full list of tags available somewhere? or would you consider sharing them? I'm very interested in the way you set up the system and want to look at a working example :)

**Reply by avtfaciam** [reply]
@brimwats I realise this is over two months late but I just saw this by chance, so sorry for this! But in case you're still curious, I have a list of my most relevant tags (not a full list admittedly) in my Publish site: https://brain.lkadre.com/%C2%A7+003+KEY

**Comment by avtfaciam** [comment]
For some ideas on different hierarchies, these are the ones I currently use (some could do with better names, I know):

- up / similar / down
- broader / related / narrower
- type / same-type / items
- jurisdiction / others / of-jurisdiction
- signatories / other-signatories / treaties
- court / others / decisions
- movement / peers / personalities
- author / other-works / works

I also managed to make a hierarchy for Periodic notes through
- ysemester / other-months / months 
- month / other-days / days

And for school I have
- semester / others / classes

**Reply by SkepticMystic** [reply]
I really like the school hierarchy, wishing I'd been using that 😅
